plugins/modules/vmware_vsan_hcl_db.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2023, Hewlett Packard Enterprise Development LP
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: vmware_vsan_hcl_db
+short_description: Manages the vSAN Hardware Compatibility List (HCL) database
+description:
+  - Manages vSAN HCL db on vSphere
+  - DB file can be downloaded from https://partnerweb.vmware.com/service/vsan/all.json
+author:
+  - Philipp Fruck (@p-fruck)
+requirements:
+  - vSAN Management SDK, which needs to be downloaded from VMware and installed manually.
+options:
+  source:
+    description:
+      - The path to the HCL db file
+    type: str
+    required: true
+extends_documentation_fragment:
+  - community.vmware.vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Fetch HCL db file
+  ansible.builtin.get_url:
+    url: https://partnerweb.vmware.com/service/vsan/all.json
+    dest: hcl_db.json
+    force: yes
```

Comment:
```suggestion
    force: true
```

We should use `true` instead of `yes` here (see #1660).